### PR TITLE
Rework of Lots of Puppet 5 facter additions

### PR DESCRIPTION
**Pull request by Dan33l**
Rework of original PR #86:
 * rebase
 * conflict resolution

### CODE REVIEW

File: facts/Vagrantfile [hunk 1]
```diff
@@ -7,6 +7,7 @@ VAGRANTFILE_API_VERSION = "2"
 Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 
   config.vm.hostname = "foo.example.com"
+  config.vm.synced_folder ".", "/vagrant", type: "virtualbox"
```
Comment (raphink) [inline facts/Vagrantfile]: What is the goal of syncing the whole dir?

Comment (Dan33l) [reply facts/Vagrantfile]: I pushed a new `Vagrantfile` in a separate commit since i don't know how to preserve author in this situation.

**Comment by Dan33l** [comment]
thank you.

**Comment by raphink** [comment]
thank *you* for fixing that :smile:

**Pull request merged**